include/revng/Support/GraphAlgorithms.h

```diff
@@ -537,3 +537,15 @@ nodesBetweenNewReverse(GraphT Source,
                                                                    Destination,
                                                                    IgnoreL);
 }
+
+template<class GraphT>
+bool isDAG(GraphT Graph) {
```

Comment:
I changed this in 7b60c49bba5376fbe0c3de3d0f15e2c1c810e9c7, but I have a question on this point. In this PR, `GraphT` is used with actual type always as a pointer (e.g., `mlir::Block *`). Is it good-practice to use `const GraphT &` as signature in this case (it may result confusing from an user point of view), or should we do something different like restricting the input template type with something like `requires std::is_pointer_v<GraphT>`?

Reply:
You're right. Using `const GraphT &` is a bad choice. Let's revert back to just `GraphT`.

One day we'll have the concept `HasGraphTraits`, where we can also add something to assert that the size is `<= sizeof(intptr_t)`.

Reply:
One day! Reverted back to the old situation in 0ff3e6296c9aecc1bd81bf7c2532dfcbabd8fdfd.